### PR TITLE
[SPARK] reorganise docs

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
* reorganise welcome page for spark documentation by moving content to `main concepts` page
* split configuration page to multiple pages

### CODE REVIEW

File: docs/integrations/spark/configuration/airflow.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    dataproc_pyspark_properties={
+      "spark.extraListeners": "io.openlineage.spark.agent.OpenLineageSparkListener",
+      "spark.jars.packages": "io.openlineage:openlineage-spark:1.0.0+",
+      "spark.openlineage.transport.url": f"{openlineage_url}/api/v1/namespaces/{openlineage_namespace}/jobs/dump_orders_to_gcs/runs/{{{{lineage_run_id(run_id, task)}}}}?api_key={api_key}",
+      "spark.openlineage.namespace": openlineage_namespace,
+      "spark.openlineage.parentJobName": job_name,
+      "spark.openlineage.parentRunId": f"{{{{lineage_run_id(run_id, task)}}}}
+    },
+    dag=dag)
+```
```
Comment (kacpermuda) [inline docs/integrations/spark/configuration/airflow.md:30]: This Operator is deprecated, so we should update thix example. It's not as easy as changing some parameters name, so I'll add that to my TODO list. Maybe it's worth mentioning here that this code might be outdated?

File: docs/integrations/spark/installation.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-sidebar_position: 1
+sidebar_position: 2
```
Comment (kacpermuda) [inline docs/integrations/spark/installation.md:2]: Shouldn't the installation be at the very top? I thought the rest relies on it heavily 😄

**Review by kacpermuda — APPROVED** [review]
Just left some minor thoughts, overall good to go.

File: docs/integrations/spark/configuration/airflow.md (hunk 1 above)
Comment (kacpermuda) [reply docs/integrations/spark/configuration/airflow.md:30]: It should still work tho, so it's not crucial to mention it :)

File: docs/integrations/spark/main_concept.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+---
+sidebar_position: 1
+title: Main Concepts
```
Comment (kacpermuda) [inline docs/integrations/spark/main_concept.md:3]: Not sure about the title, but have no alternative idea.

Comment (pawel-big-lebowski) [reply docs/integrations/spark/main_concept.md:3]: Merging for now. How about `Architecture` ?

**Pull request merged**